### PR TITLE
platform checks: rework exclude mechanism

### PR DESCRIPTION
**Pull request by nrainer-materialize**
https://github.com/MaterializeInc/materialize/pull/22682 does not work as expected.

See discussion in https://materializeinc.slack.com/archives/C057X0NG91B/p1698584377044219.

### CODE REVIEW

File: misc/python/materialize/checks/checks.py [hunk 1]
```diff
@@ -26,6 +25,15 @@ def __init__(self, base_version: MzVersion, rng: Random | None) -> None:
     def _can_run(self, e: Executor) -> bool:
         return True
 
+    def _is_enabled(self) -> tuple[bool, str | None]:
```
Comment (nrainer-materialize) [inline misc/python/materialize/checks/checks.py:28]: I kept it simple; alternatively, I could also introduce something like `YesIgnore(reason)` and `NoIgnore()` as in the output-consistency test framework.

Comment (def-) [reply misc/python/materialize/checks/checks.py:28]: Does this work? tuple[False, str] | True

**Review by def- — APPROVED** [review]

File: misc/python/materialize/checks/checks.py (hunk 1 above)
Comment (nrainer-materialize) [reply misc/python/materialize/checks/checks.py:28]: Interesting... 😀 Let me give it a try!

Comment (nrainer-materialize) [reply misc/python/materialize/checks/checks.py:28]: Nope: `Expected type expression but received "Literal[False]"`

**Review by def- — COMMENTED** [review]
I have an alternative fix, not sure if it's preferred: https://github.com/MaterializeInc/materialize/pull/22775

**Comment by nrainer-materialize** [comment]
Closing in favor of https://github.com/MaterializeInc/materialize/pull/22775.

**Pull request closed without merging**